include/UIManager.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef UIMANAGER_HPP
+# define UIMANAGER_HPP
+
+# include <Overlay/OgreOverlayManager.h>
+# include "UIOverlay.hpp"
+# include "UIOverlayHUD.hpp"
+# include "UIOverlayMenu.hpp"
+
+class UIManager {
+
+	public:
+		UIManager(UIManager const &) = delete;
+		UIManager(UIManager &&) = delete;
+		UIManager &operator=(UIManager const &) = delete;
+		UIManager &operator=(UIManager &&) = delete;
+		~UIManager(void) = default;
+		
+		static void init();
+		
+		static void showOverlayByName(std::string const &);
+		static void hideOverlayByName(std::string const &);
+		static void hideAllOverlays(void);
+		static UIOverlay *getByName(Ogre::String const &);
```

Comment:
Why return a pointer and not a reference?
When interacting with the Ogre API it's understandable, but in the general context this is not acceptable unless you have a good rational.